docs: restore metadata and docs (#1693)
jbsparrow/CyberDropDownloader: CONTRIBUTING.md

### Submitting a Pull Request (PR)

Before you submit your Pull Request (PR) consider the following guidelines:

- Search the [repository](https://github.com/Cyberdrop-DL/cyberdrop-dl/pulls) for an open or closed PR
  that relates to your submission. You don't want to duplicate effort.
- Clone the repo and make your changes on a \***\*new branch\*\*** in your fork. The base branch should be `main`
- Follow [code style conventions](#code-style)
- Commit your changes using a descriptive commit message
- Push your fork to GitHub
- In GitHub, create a pull request to the `main` branch of the repository.
- Add a description to your PR. If the PR is small (such as a typo fix), you can go brief. Sometime the title of the PR is enough.
  If it contains a lot of changes, it's better to write more details.
  If your changes are user-facing (e.g. a new feature in the UI, a change in behavior, or a bugfix)
  please include a short message to add to the changelog.
- Wait for a maintainer to review your PR and then address any comments they might have.

If everything is okay, your changes will be merged into the project.

## Setting up the development environment

> 1. Install `uv` (<https://docs.astral.sh/uv/getting-started/installation/>) (project management).

Author: ```suggestion
1. Install `uv` for project management (<https://docs.astral.sh/uv/getting-started/installation/>).
```

> 1. Install `uv` (<https://docs.astral.sh/uv/getting-started/installation/>), the project management package Cyberdrop-DL uses. `uv` will automatically install a compatible python (if required)

Author: ```suggestion
1. Install `uv` (<https://docs.astral.sh/uv/getting-started/installation/>) (project management).
```

2. Clone the repo

   ```sh
   git clone "https://github.com/Cyberdrop-DL/cyberdrop-dl"
   cd cyberdrop-dl
   ```

> 3. Setup the dev enviroment with `uv`

Author: ```suggestion
3. Setup the dev enviroment with `uv`. It will automatically install a compatible python version (if required)
```

>    uv sync --all-extras

Author: ```suggestion
   
   ```sh
   uv sync --all-extras
```

>    ```shell

Author: ```suggestion
   ```sh
```

   ```

4. Install the pre-commit hooks:

>    ```shell

Author: ```suggestion
   ```sh
```

   uv run prek install
   ```

## Code Style

### Standards

`Formatting`: This project uses [ruff](https://docs.astral.sh/ruff) for formatting, linting and import sorting.

> `TypeChecking`: Type Checking is not enforced but highly recommended. The project includes config for [`basedpyright`](https://docs.basedpyright.com/latest)

Author: ```suggestion
`Type checking`: Not enforced but highly recommended. The project includes hardcoded config options for [`basedpyright`](https://docs.basedpyright.com/latest)
```

> `TypeChecking`: Type Checking is not enforced but highly recommended. The projects includes config for [`basedpyright`](https://docs.basedpyright.com/latest)

Author: ```suggestion
`TypeChecking`: Type Checking is not enforced but highly recommended. The project includes config for [`basedpyright`](https://docs.basedpyright.com/latest)
```

> `Line Width`: We use a line width of 120.

Author: ```suggestion
`Line width`: We use a line width of 120.
```

### Code formatting with pre-commit hooks

> This project uses git pre-commit hooks to perform formatting and linting before a commit is allowed,

Author: ```suggestion
This project uses git pre-commit hooks to perform formatting and linting before a commit is created,
```

to ensure consistent style and catch some common issues early on.

Once installed, hooks will run when you commit. If the formatting isn't quite right or a linter catches something,
the commit will be rejected and `ruff` will try to fix the files. If `ruff` can not fix all the issues,
you will need to look at the output and fix them manually. When everything is fixed (either by `ruff` itself or manually)
all you need to do is `git add` those files again and retry your commit.

### Manual code formatting

We recommend [setting up your IDE](https://docs.astral.sh/ruff/editors/) to format and check with `ruff`, but you can always run
`uv run ruff check --fix` then `uv run ruff format` in the root directory before submitting a pull request.

> If you're using VScode, you can set it to [auto format python files with ruff on save](#editor-settings) in your `settings.json`

Author: ```suggestion
If you're using VScode, you can set it to [auto format python files with ruff on save](#editor-settings) in your `settings.json`

The project includes predefined [settings for the zed editor](https://github.com/Cyberdrop-DL/cyberdrop-dl/tree/main/.zed), with a recipe to quickly launch CDL in debug mode for testing
```
